### PR TITLE
Support blocking user accounts

### PR DESCRIPTION
**Pull request by tsauerwein**
Related to https://github.com/c2corg/v6_api/issues/484

Requires https://github.com/c2corg/pydiscourse/pull/1

### CODE REVIEW

File: c2corg_api/tests/views/test_user_block.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+            'user_id': self.contributor.id
+        }
+
+        headers = self.add_authorization_header(username='moderator')
+        self.app_post_json(
+            self._prefix, request_body, status=200, headers=headers)
+
+        self.assertFalse(self.is_blocked(self.contributor.id))
+
+    def test_follow_invalid_user_id(self):
```
Comment (asaunier) [inline c2corg_api/tests/views/test_user_block.py]: Do you mean ``test_unblock_invalid_user_id``?

File: c2corg_api/tests/views/test_user_block.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        self.assertTrue(self.is_blocked(self.contributor2.id))
+
+
+class TestUserBlockedRest(BaseBlockTest):
+
+    def setUp(self):  # noqa
+        super(TestUserBlockedRest, self).setUp()
+        self._prefix = '/users/blocked'
+
+    def test_follow_unauthorized(self):
```
Comment (asaunier) [inline c2corg_api/tests/views/test_user_block.py]: test_blocked_unauthorized

File: c2corg_api/tests/views/test_user_block.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        response = self.app.get(
+            self._prefix + '/invalid-user-id',
+            status=400, headers=headers)
+
+        body = response.json
+        self.assertEqual(body.get('status'), 'error')
+        errors = body.get('errors')
+        self.assertIsNotNone(self.get_error(errors, 'id'))
+
+    def test_following_wrong_user_id(self):
```
Comment (asaunier) [inline c2corg_api/tests/views/test_user_block.py]: test_blocked_wrong_user_id

File: c2corg_api/tests/views/test_user_block.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+        self.app.get(self._prefix + '/9999999999', status=400, headers=headers)
+
+
+class TestUserBlockedAllRest(BaseBlockTest):
+
+    def setUp(self):  # noqa
+        super(TestUserBlockedAllRest, self).setUp()
+        self._prefix = '/users/blocked'
+
+    def test_follow_unauthenticated(self):
```
Comment (asaunier) [inline c2corg_api/tests/views/test_user_block.py]: test_blocked_unauthenticated

**Comment by tsauerwein** [comment]
Thanks for the review.

**Comment by asaunier** [comment]
First PR with an Alembic job to run in demo/prod :P

**Comment by tsauerwein** [comment]
> First PR with an Alembic job to run in demo/prod :P

Yes! I think in our current *upgrade demo/prod* procedure we are not yet running an Alembic upgrade. But this should not be a problem, @arnaud-morvan?

**Comment by asaunier** [comment]
The wiki explains how to make an alembic upgrade in the demo:
https://github.com/c2corg/v6_api/wiki/Upgrade-the-exoscale-demo-server
```
docker-compose run --rm api .build/venv/bin/alembic upgrade head
```

No mention in the prod upgrade howto but I guess it's the same?
https://github.com/c2corg/v6_api/wiki/Upgrade-the-prod-server

For a dev env, I guess one should simply run
```
.build/venv/bin/alembic upgrade head
```
as explained at https://github.com/c2corg/v6_api/tree/master/alembic_migration#run-a-migration
Right?

**Comment by arnaud-morvan** [comment]
I've added the alembic migration to prod upgrade wiki page.

**Pull request merged**